snap/file.go

```diff
@@ -43,6 +38,22 @@ type File interface {
 	// NeedsMountUnit determines whether it's required to setup
 	// a mount unit for the snap when the snap is installed
 	NeedsMountUnit() bool
+
+	// Info returns information about the given snap file
+	Info() (*Info, error)
+}
+
+// Backend implements a specific snap format
+type Backend struct {
```

Comment:
Looks like this should be private for now. I'd also take the chance we have a good term for it after "file format" (e.g. "snapFormat" and "RegisterFormat", or something similar of your liking) and reserve the generic "backend" wildcard term for something else.